### ISSUE TITLE
triggerHook position changes when mobile address bar opens 

### ISSUE THREAD

**Issue by soniagp6** [janpaepke/ScrollMagic#883]
Thank you for your wonderful plugin. It's been immensely useful to me through several projects!

I'm testing a full-viewport-height parallax element on mobile and noticed that the triggerHook (positioned at the bottom of the viewport) pops down when the bottom bar of the mobile browser is hidden, causing a quick pop in the parallax animation. This seems to be due to the fact that it has position: fixed which is the one type of positioning still affected when mobile browser navbars appear and disappear: https://developers.google.com/web/updates/2016/12/url-bar-resizing (I think? This is an older post but as far as I can tell things haven't changed). 

Two screenshots to illustrate the behavior
https://pasteboard.co/I9nwLhT.png
https://pasteboard.co/I9nxeJw.png

Is there any workaround to get the triggerHook to stay at the bottom of the viewport, in spite of the presence of the bottom bar? 

Thank you so much (in advance)!!

**Comment by soniagp6** [comment]
Just as an extra precaution, I took a look at this on /examples/basic/scene_manipulation.html tutorial on my phone, switching the triggerhook position to 1, and saw the same behavior with the bottom bar. I realize for most types of animations this would be the preferred behavior, it's just  in this specific case that it's giving me difficulties...

**Comment by jperkinsx** [comment]
Did you ever find a fix for this?

I have a related issue: a page with three pinned scenes that appear, play and go by as you scroll down, with additional static content below them.

On mobile browsers, when you change scroll direction and show/hide the browser's url bar, the `padding` of  the `scrollmagic-pin-spacer` is automatically recalculated based on the new window height.

The consequence of this is that everything beneath the first scene jumps up and down, as it is pushed by the change in padding height. 

Not sure how to fix this, any ideas appreciated!

**Comment by dividerbeam** [comment]
I was fix it with "ios-inner-height" (https://github.com/tylerjpeterson/ios-inner-height)

Try to change function getViewportSize() in ScrollMagic.js

You should replace "_util.get.height(_options.container)" on "iosInnerHeight()"
and in all your scene scripts, where you use window.innerHeight, replace it on "iosInnerHeight()"

**Comment by janpaepke** [comment]
Sorry for the very long wait on this. The mobile address bar resizing issue was a known pain point with v2 — it used `window.innerHeight` for viewport calculations, which changes when the browser chrome appears or disappears, causing trigger positions to shift. The workaround from @dividerbeam using `ios-inner-height` was a reasonable v2 fix.

v3 uses IntersectionObserver for trigger detection, which the browser evaluates against the actual visible viewport. Modern browsers (particularly iOS Safari 15+) also now support `dvh`/`svh`/`lvh` viewport units and `visualViewport` API, making this less of a minefield. That said, mobile viewport behavior remains inherently tricky — if you hit edge cases with v3, please file a new issue.

---
**ScrollMagic v3 is now available** — a complete rewrite with an entirely new architecture and API. v2 will not receive further updates (2.0.9 is the final release).

👉 `npm install scrollmagic@next` — [Docs](https://github.com/janpaepke/ScrollMagic/blob/main/README.md) · [Release notes](https://github.com/janpaepke/ScrollMagic/releases/tag/v3.0.0-beta.0) · [npm](https://www.npmjs.com/package/scrollmagic/v/3.0.0-beta.0)
